lockable-object/src/main/java/com/iluwatar/lockableobject/App.java

```diff
@@ -0,0 +1,67 @@
+package com.iluwatar.lockableobject;
+
+import com.iluwatar.lockableobject.domain.Creature;
+import com.iluwatar.lockableobject.domain.Elf;
+import com.iluwatar.lockableobject.domain.Feind;
+import com.iluwatar.lockableobject.domain.Human;
+import com.iluwatar.lockableobject.domain.Orc;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The Lockable Object pattern is a concurrency pattern. Instead of using the "synchronized" word
+ * upon the methods to be synchronized, the object which implements the Lockable interface handles
+ * the request.
+ *
+ * @author Noam Greenshtain
+ */
+public class App implements Runnable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(App.class.getName());
+  private static final int WAIT_TIME = 2;
+  private static final int WORKERS = 2;
+  private static final int MULTIPLICATION_FACTOR = 3;
+
+  /**
+   * main method.
+   *
+   * @param args as arguments for the main method.
+   */
+  public static void main(String[] args) {
+    var app = new App();
+    app.run();
+  }
+
+  @Override
+  public void run() {
+    Lockable sword = new SwordOfAragorn();
```

Comment:
We can use the local variable inference `var` syntax. Please check this elsewhere too.

Reply:
It would be nice to have some high-level comments, what's happening